Lib/asyncio/base_events.py

```diff
@@ -147,6 +149,39 @@ def _ipaddr_info(host, port, family, type, proto):
     return None
 
 
+def _roundrobin(*iterables):
+    """roundrobin('ABC', 'D', 'EF') --> A D E B F C"""
+    # Copied from Python docs, Recipe credited to George Sakkis
+    pending = len(iterables)
+    nexts = itertools.cycle(iter(it).__next__ for it in iterables)
+    while pending:
+        try:
+            for next in nexts:
+                yield next()
+        except StopIteration:
+            pending -= 1
+            nexts = itertools.cycle(itertools.islice(nexts, pending))
+
+
+def _interleave_addrinfos(addrinfos, first_address_family_count=1):
+    """Interleave list of addrinfo tuples by family."""
+    # Group addresses by family
+    addrinfos_by_family = collections.OrderedDict()
+    for addr in addrinfos:
+        family = addr[0]
+        if family not in addrinfos_by_family:
+            addrinfos_by_family[family] = []
+        addrinfos_by_family[family].append(addr)
+    addrinfos_lists = list(addrinfos_by_family.values())
+
+    reordered = []
+    if first_address_family_count > 1:
+        reordered.extend(addrinfos_lists[0][:first_address_family_count - 1])
+        del addrinfos_lists[0][:first_address_family_count - 1]
+    reordered.extend(_roundrobin(*addrinfos_lists))
+    return reordered
+
+
```

Comment:
`base_events.py` is already overly long. Would it make sense to move these two functions to `staggered.py` (keeping them private)?